### PR TITLE
75189 change validation for vaMedicalFacility

### PR DESCRIPTION
**Pull request by jadon1979**
## Summary

Currently, vets-json-schema has `vaMedicalFacility` as an enum type.  This causes issues on validations if the Frontend 
is using Lighthouse as there are more facilities in that list than in the enum.  The update here solves the issue while we 
remove the code from vets-website and vets-json-schema.  

## Related issue(s)

[https://app.zenhub.com/workspaces/10-10-health-apps-5fff0cfd1462b6000e320fc7/issues/gh/department-of-veterans-affairs/va.gov-team/75189](https://app.zenhub.com/workspaces/10-10-health-apps-5fff0cfd1462b6000e320fc7/issues/gh/department-of-veterans-affairs/va.gov-team/75189)

## Testing done

- manual and automated

## What areas of the site does it impact?

- 1010EZ submissions where `hca_use_facilities_API` is enabled.

### CODE REVIEW

File: app/models/health_care_application.rb [hunk 1]
```diff
@@ -267,4 +267,28 @@ def log_submission_failure
...
+      JSON::Validator.fully_validate(current_schema, parsed_form).each do |v|
+        errors.add(:form, v.to_s)
+      end
+    end
+  end
+
+  def current_schema
+    @current_schema ||= begin
+      schema = VetsJsonSchema::SCHEMAS[self.class::FORM_ID]
+      if Flipper.enabled?(:hca_use_facilities_API)
```
Comment (jadon1979) [inline app/models/health_care_application.rb:285]: using unless with a return would have been cleaner, however rubocop was unhappy with this and adding an override here was frankly more problematic in the future... potentially.

**Review by lihanli — APPROVED** [review]

**Review by JoshingYou1 — APPROVED** [review]
LGTM!

File: app/models/health_care_application.rb (hunk 1 above)
Comment (Mitch-A6) [reply app/models/health_care_application.rb:285]: 🤔 Do you remember which rubocop complaint it was? `return [...] unless [...]` is such a common ruby pattern, that's a surprise to me.

I guess the only reasonable alternative here would be putting 284-290 into _another_ function that early-returns on its own, but that does seem excessive.

File: spec/requests/health_care_applications_request_spec.rb [hunk 2]
```diff
@@ -408,6 +408,32 @@ def self.expect_async_submit
...
+        end
+
+        let(:params) do
+          test_veteran['vaMedicalFacility'] = '000'
+          {
+            form: test_veteran.to_json
+          }
+        end
+
+        let(:body) do
```
Comment (Mitch-A6) [inline spec/requests/health_care_applications_request_spec.rb:424]: 🤔 I get the use of `let` for `params` so the override can go in to the `subject` definition, but is there a reason to use it for `body` vs defining it locally inside the spec?

**Pull request merged**